### PR TITLE
NO-JIRA: test/e2e: detect rapidly updating deployments

### PR DESCRIPTION
**Pull request by sjenning**

### CODE REVIEW

File: test/e2e/util/util.go [hunk 1]
```diff
@@ -548,6 +548,23 @@ func NoticePreemptionOrFailedScheduling(t *testing.T, ctx context.Context, clien
 	})
 }
 
+func EnsureNoRapidDevelopmentRollouts(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	const maxAllowedGeneration = 30
```
Comment (enxebre) [inline test/e2e/util/util.go:552]: isn't this too permissive?

Comment (sjenning) [reply test/e2e/util/util.go:552]: Probably, I just didn't want false positives. Looking at deployments from `TestUpgradeControlPlane`, seems the highest generation is `4`.  Could tighten it to 10 to be sensitive.

**Comment by sjenning** [comment]
Updated PR:
- `s/EnsureNoRapidDevelopmentRollouts/EnsureNoRapidDeploymentRollouts`
- tightened `maxAllowedGeneration` to `10`

**Pull request merged**